core/core-micro-state.el

```diff
@@ -265,4 +265,63 @@ pressed)."
       (delete-window corelv-wnd)
       (kill-buffer buf))))
 
+(defmacro spacemacs|define-micro-state-2 (name &rest props)
+  "Define a micro-state called NAME.
+NAME is a symbol.
+Available PROPS:
+`:on-enter SEXP'
+    Evaluate SEXP when the micro-state is switched on.
+`:on-exit SEXP'
+    Evaluate SEXP when leaving the micro-state.
+`:doc STRING or SEXP'
+    A STRING or a SEXP that evaluates to a string.
+`:columns INTEGER'
+    Automatically generate :doc with this many number of columns.
+`:hint BOOLEAN'
+    Whether to automatically add hints to the docstring. Default is nil.
+`:persistent BOOLEAN'
+    If BOOLEAN is non nil then the micro-state never exits. A binding
+    with an explicitly set `exit t' property is required. Default is nil.
+`:foreign-keys SYMBOL'
+    What to do when keys not bound in the micro-state are entered. This
+    can be nil (default), which means to exit the micro-state, warn,
+    which means to not exit but warn the user that the key is not part
+    of the micro-state, or run, which means to try to run the key binding
+    without exiting.
+`:entry-binding MAP KEY'
+    Key binding to use for entering the micro-state.
+`:bindings EXPRESSIONS'
+    One or several EXPRESSIONS with the form
+    (STRING1 SYMBOL1 DOCSTRING
+                     :exit SYMBOL)
+    where:
+    - STRING1 is a key to be bound to the function or key map SYMBOL1.
+    - DOCSTRING is a STRING or an SEXP that evaluates to a string
+    - :exit SYMBOL or SEXP, if non nil then pressing this key will
+      leave the micro-state (default is nil).
+      Important note: due to inner working of transient-maps in Emacs
+      the `:exit' keyword is evaluate *before* the actual execution
+      of the bound command.
+All properties supported by `spacemacs//create-key-binding-form' can be
+used."
```

Comment:
Would be cool to support this as well see line 113. This is for quick binding of the micro-state and this is usable in both micro-states and toggles.